src/client.js

```diff
@@ -30,13 +30,16 @@ class Client extends GeneratedApiClient {
   constructor(clientConfig) {
     super();
     const configLoader = new ConfigLoader();
+    clientConfig = clientConfig || {};
```

Comment:
IMO -  Reassigning function params is a bad practice. Can we add the [`no-param-reassign`](https://eslint.org/docs/rules/no-param-reassign) rule to our eslint config?

Reply:
maybe just use a default param?

Reply:
Good catch, I do like the idea of enforcing that rule. It does mean a lot of changes for this lib though (we do this in about 20 places).  I'm going to fix this instance but will put off adding that rule for a refactor PR